source/h2_decoder.c

```diff
@@ -756,24 +732,14 @@ static int s_state_fn_frame_continuation(struct aws_h2_decoder *decoder, struct
     return aws_raise_error(AWS_ERROR_UNIMPLEMENTED);
 }
 
-static int s_state_fn_frame_custom(struct aws_h2_decoder *decoder, struct aws_byte_cursor *input) {
+static int s_state_fn_frame_unknown(struct aws_h2_decoder *decoder, struct aws_byte_cursor *input) {
 
     const uint32_t remaining_length = decoder->frame_in_progress.payload_len;
 
-    struct aws_byte_cursor data_to_pass;
-    bool will_finish_state;
     if (input->len < remaining_length) {
-        data_to_pass = aws_byte_cursor_advance(input, input->len);
-        will_finish_state = false;
+        aws_byte_cursor_advance(input, input->len);
     } else {
-        data_to_pass = aws_byte_cursor_advance(input, remaining_length);
-        will_finish_state = true;
-    }
-    AWS_FATAL_ASSERT(data_to_pass.len <= UINT32_MAX);
-
-    DECODER_CALL_VTABLE_STREAM_ARGS(decoder, on_custom_frame_payload, &data_to_pass);
-
-    if (will_finish_state) {
+        aws_byte_cursor_advance(input, remaining_length);
```

Comment:
don't we need to decrement `decoder->frame_in_progress.payload_len`?

Reply:
Not here, because we're resetting the state immediately after this, but for the first case, yes, yes we should.